File: Data Structures & Algorithms/maximum-product-subarray/submission-9.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        maxEnding, minEnding = 1, 1 
        res = float('-inf') 

        for num in nums:  
            curMaxEnding = maxEnding
            maxEnding = max(maxEnding * num, minEnding * num, num)  
            minEnding = min(minEnding * num, curMaxEnding * num, num) 

            res = max(maxEnding, minEnding, res) 

        return res

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-9.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let set = new Set(nums);
        let max = 0;

        for (let num of nums) {
            if (set.has(num - 1)) continue;

            let total = 0; 
            while (set.has(num)) {
                total ++
                num = num + 1
            }

            max = Math.max(max, total)
        }

        return max
    }
}
